filters/auth/grantprep.go

```diff
@@ -0,0 +1,33 @@
+package auth
+
+import "github.com/zalando/skipper/eskip"
+
+const (
+	defaultCallbackRouteID = "__oauth2_grant_callback"
+	defaultCallbackPath    = "/.well-known/oauth2-callback"
+	defaultTokenCookieName = "oauth-grant"
+)
+
+type grantPrep struct {
+	config OAuthConfig
+}
+
+func (p *grantPrep) Do(r []*eskip.Route) []*eskip.Route {
```

Comment:
Add godoc:
```
Do adds automatically a global callback route to the routes
```

Reply:
Is it necessary? the grantPrep is not an exported type. The right place to document what it does would be godoc for the `OAuthConfig.NewGrantPreprocessor` method.

Reply:
ok